runners/apex/src/main/java/org/apache/beam/runners/apex/translation/TranslationContext.java

```diff
@@ -142,16 +147,19 @@ public void addOperator(Operator operator, OutputPort port, PCollection output)
       name = getCurrentTransform().getFullName() + i;
     }
     this.operators.put(name, operator);
-    this.streams.put(output, (Pair) new ImmutablePair<>(port, new ArrayList<>()));
+    this.streams.put(output, (Pair) new ImmutablePair<>(
+            new OutputPortInfo(port, getCurrentTransform()),
+            new ArrayList<>()));
   }
 
   public void addStream(PInput input, InputPort inputPort) {
     while (aliasCollections.containsKey(input)) {
       input = aliasCollections.get(input);
     }
-    Pair<OutputPort<?>, List<InputPort<?>>> stream = this.streams.get(input);
-    checkArgument(stream != null, "no upstream operator defined for %s", input);
-    stream.getRight().add(inputPort);
+
+    Pair<OutputPortInfo, List<InputPortInfo>> stream = this.streams.get(input);
+    checkArgument(stream != null, "no upstream operator defined for " + input);
```

Comment:
YARN launcher was failing at this change... I think in my copy of beam. Now it is fine. Reverted this change.